core/templates/dev/head/domain/state_card/StateCardObjectFactory.js

```diff
@@ -17,114 +17,189 @@
-    if (this.getLastInputResponsePair() === null) {
-      return null;
-    }
-    return this.getLastInputResponsePair().oppiaResponse;
-  };
-
-  StateCard.prototype.addInputResponsePair = function(inputResponsePair) {
-    this._inputResponsePairs.push(angular.copy(inputResponsePair));
-  };
-
-  StateCard.prototype.setDestStateName = function(destStateName) {
-    this._destStateName = destStateName;
-  };
-
-  StateCard.prototype.setOppiaResponse = function(index, response) {
-    this._inputResponsePairs[index].oppiaResponse = response;
-  };
-
-  StateCard.prototype.setLastOppiaResponse = function(response) {
-    this.setOppiaResponse(this._inputResponsePairs.length - 1, response);
-  };
-
-  StateCard.prototype.setInteractionHtml = function(interactionHtml) {
-    this._interactionHtml = interactionHtml;
-  };
-
-  StateCard.prototype.setLeadsToConceptCard = function(leadsToConceptCard) {
-    this._leadsToConceptCard = leadsToConceptCard;
-  };
-
-  /**
-   * @param {string} stateName - The state name for the current card.
-   * @param {object} params - The set of parameters for the learner associated
-   *        with a card.
-   * @param {string} contentHtml - The HTML string for the content displayed on
-   *        the content card.
-   * @param {string} interactionHtml - The HTML that calls the interaction
-   *        directive for the current card.
-   * @param {Interaction} interaction - An interaction object that stores all
-   *        the properties of the card's interaction.
-   * @param {bool} leadsToConceptCard - Whether the current card leads to a
-   *        concept card in the exploration.
-   */
-  StateCard.createNewCard = function(
-      stateName, params, contentHtml, interactionHtml, interaction,
-      leadsToConceptCard) {
-    return new StateCard(
-      stateName, params, contentHtml, interactionHtml, interaction,
-      leadsToConceptCard, null, []);
-  };
-
-  return StateCard;
-}]);
+    };
+
+    StateCard.prototype.isCardTerminal = function() {
+      var interactionId = this.getInteractionId();
+      return (
+        interactionId && INTERACTION_SPECS[interactionId].is_terminal);
+    };
+
+    StateCard.prototype.getInteractionInstructions = function() {
+      var interactionId = this.getInteractionId();
+      return (
+        interactionId ? INTERACTION_SPECS[interactionId].instructions : '');
+    };
+
+    StateCard.prototype.getInteractionCustomizationArgs = function() {
+      var interaction = this.getInteraction();
+      if (!interaction) {
+        return null;
+      }
+      return interaction.customizationArgs;
+    };
+
+    StateCard.prototype.isInteractionInline = function() {
+      var interactionId = this.getInteractionId();
+      return (
+        !interactionId ||
+        INTERACTION_SPECS[interactionId].display_mode ===
+          INTERACTION_DISPLAY_MODE_INLINE);
+    };
+
+    StateCard.prototype.getCurrentParams = function() {
```

Comment:
"Current" seems like an indicator that this shouldn't be part of StateCard.

Reply:
Yeah, this needn't be in state card, will remove it.

Reply:
No, params are not shown to the learner. Probably more relevant for the player state service or the conversation skin service -- it's something that that service can call LearnerParamsService to update at the right times. Which of those two services to put it in depends on when exactly the param transition should happen.

Reply:
Done, removed params from card object.